src/pages/subject/create/index.tsx

```diff
@@ -82,12 +80,12 @@ function CreateUserPage() {
     return (
       <form
         onSubmit={handleSubmit(async (data) => {
-          const baseData = { permissions: selectedPermissions }
+          const permissions = data.permissions.map((permission) => fetchPermissionName(permission, permissionsListData))
```

Comment:
We're making an api call inside a `.map` also inside a the `handleSubmit`? Is this stable? What if the api request fails?